File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        Map<Character, Integer> letters = new HashMap<>();
        if(s.length() != t.length())
        {
            return false;
        }
        for (char c : s.toCharArray())
         {
           letters.put(c, letters.getOrDefault(c, 0) + 1);
         }
        for(char c : t.toCharArray())
        {
         if(letters.containsKey(c))
         {
            letters.put(c, letters.get(c) - 1);
            if(letters.get(c) == 0)
            {
                letters.remove(c);
            }
         }
         else
         {
            return false;
         }
        }
        return true;
    }
}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        seen = {}
        for char in s:
            freq = seen.get(char, 0)
            seen[char] = freq+1
        for char in t:
            freq = seen.get(char, 0)
            seen[char] = freq-1
        
        for val in seen.values():
            if val != 0:
                return False
        return True
